Mage/src/main/java/mage/abilities/decorator/ConditionalEvasionEffect.java

```diff
@@ -0,0 +1,126 @@
+        this.baseCondition = condition;
+        this.otherwiseEffect = otherwiseEffect;
+        if (text != null) {
+            this.setText(text);
+        }
+    }
+
+    public ConditionalEvasionEffect(final ConditionalEvasionEffect effect) {
+        super(effect);
+        this.effect = (EvasionEffect) effect.effect.copy();
+        if (effect.otherwiseEffect != null) {
+            this.otherwiseEffect = (EvasionEffect) effect.otherwiseEffect.copy();
+        }
+        this.condition = effect.condition;
+        this.conditionState = effect.conditionState;
+        this.baseCondition = effect.baseCondition;
+        this.initDone = effect.initDone;
+    }
+
+    @Override
+    public void init(Ability source, Game game) {
+        super.init(source, game);
+        if (baseCondition instanceof LockedInCondition) {
+            condition = new FixedCondition(((LockedInCondition) baseCondition).getBaseCondition().apply(game, source));
+        } else {
+            condition = baseCondition;
+        }
+        effect.setTargetPointer(this.targetPointer);
+        effect.init(source, game);
+        if (otherwiseEffect != null) {
+            otherwiseEffect.setTargetPointer(this.targetPointer);
+            otherwiseEffect.init(source, game);
+        }
+        initDone = true;
+    }
+
+    @Override
+    public boolean applies(Permanent permanent, Ability source, Game game) {
+        if (!initDone) { // if simpleStaticAbility, init won't be called
+            init(source, game);
+        }
+        conditionState = condition.apply(game, source);
+        if (conditionState) {
+            effect.setTargetPointer(this.targetPointer);
+            return effect.applies(permanent, source, game);
+        } else if (otherwiseEffect != null) {
+            otherwiseEffect.setTargetPointer(this.targetPointer);
+            return otherwiseEffect.applies(permanent, source, game);
+        }
+        if (effect.getDuration() == Duration.OneUse) {
+            used = true;
+        }
+        if (effect.getDuration() == Duration.Custom) {
+            this.discard();
+        }
+        return false;
+    }
+
+    @Override
+    public boolean cantBeBlocked(Permanent attacker, Permanent blocker, Ability source, Game game, boolean canUseChooseDialogs) {
+        if (conditionState) {
+            return effect.cantBeBlocked(attacker, blocker, source, game, canUseChooseDialogs);
+        } else if (otherwiseEffect != null) {
+            return otherwiseEffect.cantBeBlocked(attacker, blocker, source, game, canUseChooseDialogs);
+        }
+        return true;
+    }
+
+    @Override
+    public String cantBeBlockedMessage(Permanent attacker, Ability source, Game game) {
+        if (conditionState) {
+            return effect.cantBeBlockedMessage(attacker, source, game);
+        } else if (otherwiseEffect != null) {
+            return otherwiseEffect.cantBeBlockedMessage(attacker, source, game);
+        }
+        return null;
```

Comment:
I don't like `null` here and null checking in hint's cantBeBlockedMessage result usage. It's impossible situation (see comment above about "true" return).

If you have active restriction then it must have some message all the time. If you haven't active restriction then you must not use a message at all.

So I recommends to add wrong code usage exception here instead a null result.

Reply:
The message is only used for hints, 
I guess I'll add a `boolean hasCantBeBlockedMessage` method, that if returning true, means there is a not-null message to find using  `cantBeBlockedMessage` (that I'll rename into `getCantBeBlockedMessage`).

Both of those will not actually be used in any place other than `getRules` collectings hints, as the `cantBeBlocked` method with more arguments requires a blocker and is not there to display continuously that there is an evasion effect.